dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/CaseFormats.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations.processor.data;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.base.CharMatcher;
+import com.palantir.logsafe.Preconditions;
+import java.util.Optional;
+
+/**
+ * Utility functionality for handling guava {@link CaseFormat} values.
+ */
+final class CaseFormats {
```

Comment:
I've written or fixed this code in a few projects (conjure-java, metric-schema, an internal config annotation processor). We should probably add a new oss library specifically around case formats and conversion between them... not today though.

Reply:
Not today, not today...

Reply:
ha it has been suggested by others and shut down in guava ;-)